### DISCUSSION TITLE
Ruff --preview required

### DISCUSSION THREAD

**Discussion by Mjb141** [LazyVim/LazyVim#4143] [Q&A] [answered]
Ruff complains about requiring `--preview`:

`[ERROR][2024-07-22 17:07:20] .../vim/lsp/rpc.lua:770	"rpc"	"/Users/m/.local/share/nvim/mason/bin/ruff"	"stderr"	"error: --preview needs to be provided as a command line argument while the server is still unstable.\nFor example: `ruff server --preview`\n"`

It's not clear (or I can't find) in the documentation where this additional parameter must be set. I've already got ruff enabled `opts.servers.ruff.enabled = true` and `vim.g.lazyvim_python_ruff = "ruff"` in options.lua.

**Comment by folke** [comment, marked as answer]
update ruff with mason.
